### PR TITLE
Ported 5 subtractions functions

### PR DESCRIPTION
**Pull request by NoumanAmir657**

### CODE REVIEW

File: source/vsub_s16_rvv.c [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * making it analogous to the ARM Neon function `vsub_s16`. The result is a new vector where each element 
+ * is the difference of the corresponding elements in the input vectors.
+ * 
+ * @param a The first input vector of type int16x4_t containing 16-bit signed integers.
+ * @param b The second input vector of type int16x4_t containing 16-bit signed integers.
+ * @return int16x4_t The result of the element-wise subtraction, returned as an 16-bit signed integer vector.
+ */
+int16x4_t vsub_s16_rvv(int16x4_t a, int16x4_t b) {
+    return __riscv_vsub_vv_i16m1(a, b, VLEN_4);
+}
```
Comment (mmhus) [inline source/vsub_s16_rvv.c]: There is no empty line at the end of file for all files, please fix.

File: source/vsub_s32_rvv.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * making it analogous to the ARM Neon function `vsub_s32`. The result is a new vector where each element 
+ * is the difference of the corresponding elements in the input vectors.
+ * 
+ * @param a The first input vector of type int32x2_t containing 32-bit signed integers.
+ * @param b The second input vector of type int32x2_t containing 32-bit signed integers.
+ * @return int32x2_t The result of the element-wise subtraction, returned as an 32-bit signed integer vector.
+ */
+int32x2_t vsub_s32_rvv(int32x2_t a, int32x2_t b) {
+    return __riscv_vsub_vv_i32m1(a, b, VLEN_2);
+}
```
Comment (mmhus) [inline source/vsub_s32_rvv.c]: There is no empty line at the end of file for all files, please fix.

File: source/vsub_s64_rvv.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * making it analogous to the ARM Neon function `vsub_s64`. The result is a new vector where each element 
+ * is the difference of the corresponding elements in the input vectors.
+ * 
+ * @param a The first input vector of type int64x1_t containing 64-bit signed integers.
+ * @param b The second input vector of type int64x1_t containing 64-bit signed integers.
+ * @return int64x1_t The result of the element-wise subtraction, returned as an 64-bit signed integer vector.
+ */
+int64x1_t vsub_s64_rvv(int64x1_t a, int64x1_t b) {
+    return __riscv_vsub_vv_i64m1(a, b, VLEN_1);
+}
```
Comment (mmhus) [inline source/vsub_s64_rvv.c]: There is no empty line at the end of file for all files, please fix.

File: source/vsub_s8_rvv.c [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ * is the difference of the corresponding elements in the input vectors.
+ * 
+ * @param a The first input vector of type int16x4_t containing 8-bit signed integers.
+ * @param b The second input vector of type int16x4_t containing 8-bit signed integers.
+ * @return int8x8_t The result of the element-wise subtraction, returned as an 8-bit signed integer vector.
+ */
+
+int8x8_t vsub_s8_rvv(int8x8_t a, int8x8_t b) {
+    return __riscv_vsub_vv_i8m1(a, b, VLEN_8);
+}
```
Comment (mmhus) [inline source/vsub_s8_rvv.c]: There is no empty line at the end of file for all files, please fix.

File: source/vsub_u8_rvv.c [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ * making it analogous to the ARM Neon function `vsub_u8`. The result is a new vector where each element 
+ * is the difference of the corresponding elements in the input vectors.
+ * 
+ * @param a The first input vector of type uint8x8_t containing 8-bit unsigned integers.
+ * @param b The second input vector of type uint8x8_t containing 8-bit unsigned integers.
+ * @return uint8x8_t The result of the element-wise subtraction, returned as an 8-bit unsigned integer vector.
+ */
+uint8x8_t vsub_u8_rvv(uint8x8_t a, uint8x8_t b) {
+    return __riscv_vsub_vv_u8m1(a, b, VLEN_8);
+}
```
Comment (mmhus) [inline source/vsub_u8_rvv.c]: There is no empty line at the end of file for all files, please fix.

**Review by mmhus — CHANGES_REQUESTED** [review]

File: source/vsub_s16_rvv.c (hunk 1 above)
Comment (NoumanAmir657) [reply source/vsub_s16_rvv.c]: Fixed

File: source/vsub_s32_rvv.c (hunk 2 above)
Comment (NoumanAmir657) [reply source/vsub_s32_rvv.c]: Fixed

File: source/vsub_s64_rvv.c (hunk 3 above)
Comment (NoumanAmir657) [reply source/vsub_s64_rvv.c]: Fixed

File: source/vsub_s8_rvv.c (hunk 4 above)
Comment (NoumanAmir657) [reply source/vsub_s8_rvv.c]: Fixed

File: source/vsub_u8_rvv.c (hunk 5 above)
Comment (NoumanAmir657) [reply source/vsub_u8_rvv.c]: Fixed

File: source/vsub_s16_rvv.c [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 10xEngineers
+
+#include "../common/rhal_base.h"
```
Comment (muhmuz) [inline source/vsub_s16_rvv.c]: Do not use full relative path, just include the header file only.

File: source/vsub_s32_rvv.c [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 10xEngineers
+
+#include "../common/rhal_base.h"
```
Comment (muhmuz) [inline source/vsub_s32_rvv.c]: Do not use full relative path, just include the header file only.

File: source/vsub_s64_rvv.c [hunk 8]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 10xEngineers
+
+#include "../common/rhal_base.h"
```
Comment (muhmuz) [inline source/vsub_s64_rvv.c]: Do not use full relative path, just include the header file only.

File: source/vsub_s8_rvv.c [hunk 9]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2024 10xEngineers
+
+#include "../common/rhal_base.h"
```
Comment (muhmuz) [inline source/vsub_s8_rvv.c]: Do not use full relative path, just include the header file only.

File: source/vsub_u8_rvv.c [hunk 10]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 10xEngineers
+
+#include "../common/rhal_base.h"
```
Comment (muhmuz) [inline source/vsub_u8_rvv.c]: Do not use full relative path, just include the header file only.

File: common/rhal_base.h [hunk 11]
```diff
@@ -10,11 +10,19 @@ extern "C" {
 #include <stdint.h>
```
Comment (muhmuz) [inline common/rhal_base.h]: remove "stdint" header file from here, as it is already included in common_utilities.h

File: source/vsub_s16_rvv.c (hunk 6 above)
Comment (NoumanAmir657) [reply source/vsub_s16_rvv.c]: Fixed

File: source/vsub_s32_rvv.c (hunk 7 above)
Comment (NoumanAmir657) [reply source/vsub_s32_rvv.c]: Fixed

File: source/vsub_s64_rvv.c (hunk 8 above)
Comment (NoumanAmir657) [reply source/vsub_s64_rvv.c]: Fixed

File: source/vsub_s8_rvv.c (hunk 9 above)
Comment (NoumanAmir657) [reply source/vsub_s8_rvv.c]: Fixed

File: source/vsub_u8_rvv.c (hunk 10 above)
Comment (NoumanAmir657) [reply source/vsub_u8_rvv.c]: Fixed

File: common/rhal_base.h (hunk 11 above)
Comment (NoumanAmir657) [reply common/rhal_base.h]: Fixed

**Review by mmhus — APPROVED** [review]

**Pull request merged**